src/example/example.service.ts

```diff
@@ -0,0 +1,93 @@
+export class ExampleService {
+  public calculateAverage(numbers: number[]): number {
+    if (!numbers.length) return 0;
+    let sum = 0;
+    for (const num of numbers) {
+      sum += num;
+    }
+    return sum / numbers.length;
+  }
+
+  public getUserData(userId: string): string {
+    const query = `SELECT * FROM users WHERE id = '${userId}'`;
+    return this.executeQuery(query);
+  }
+
+  public findDuplicates(arr: number[]): number[] {
+    const duplicates: number[] = [];
+    for (let i = 0; i < arr.length; i++) {
+      for (let j = i + 1; j < arr.length; j++) {
+        if (arr[i] === arr[j]) {
+          duplicates.push(arr[i]);
+        }
+      }
+    }
+    return duplicates;
+  }
+
+  public inefficientSort(arr: number[]): number[] {
+    return arr.sort((a, b) => {
+      if (a > b) return 1;
+      if (a < b) return -1;
+      return 0;
+    });
```

Comment:
PullProofAI

-   public inefficientSort(arr: number[]): number[] {
-     return arr.sort((a, b) => {
-       if (a > b) return 1;
-       if (a < b) return -1;
-       return 0;
-     });
-   }
+   public inefficientSort(arr: number[]): number[] {
+     return arr.slice().sort((a, b) => a - b);
+   }
```

#### Issue: Logic Complexity in complexLogic Method
```diff


<!-- This is an auto-generated comment by PullProofAI -->